src/execution/execute.ts

```diff
@@ -749,18 +748,18 @@ function executeField(
       // Note: we don't rely on a `catch` method, but we do expect "thenable"
       // to take a second callback for the error case.
       return completed.then(undefined, (rawError) => {
-        const error = locatedError(rawError, fieldNodes, pathToArray(path));
-        const handledError = handleFieldError(error, returnType, errors);
+        const errors = asyncPayloadRecord?.errors ?? exeContext.errors;
+        addError(rawError, fieldNodes, returnType, path, errors);
         filterSubsequentPayloads(exeContext, path, asyncPayloadRecord);
-        return handledError;
+        return null;
       });
     }
     return completed;
   } catch (rawError) {
-    const error = locatedError(rawError, fieldNodes, pathToArray(path));
-    const handledError = handleFieldError(error, returnType, errors);
+    const errors = asyncPayloadRecord?.errors ?? exeContext.errors;
+    addError(rawError, fieldNodes, returnType, path, errors);
```

Comment:
Not sure about this one, especially since `addError` throws located error.
But since it's not a public API, let's try and merge it.

Future suggestion: If we convert ExecutionContext into class it can have single method called `handleFieldError` that:
1. `locateError`
2. push error into `errors`
3. `filterSubsequentPayloads`